src/ledger/test/LedgerTestUtils.cpp

```diff
@@ -190,8 +190,30 @@ makeValid(TrustLineEntry& tl)
     }
     tl.limit = std::abs(tl.limit);
     clampLow<int64>(1, tl.limit);
-    tl.asset.type(ASSET_TYPE_CREDIT_ALPHANUM4);
-    strToAssetCode(tl.asset.alphaNum4().assetCode, "USD");
+
+    switch (tl.asset.type())
+    {
+    case ASSET_TYPE_NATIVE:
+        tl.asset.type(ASSET_TYPE_CREDIT_ALPHANUM4);
+        strToAssetCode(tl.asset.alphaNum4().assetCode, "USD");
+        break;
+    case ASSET_TYPE_CREDIT_ALPHANUM4:
+        strToAssetCode(tl.asset.alphaNum4().assetCode, "USD");
+        break;
+    case ASSET_TYPE_CREDIT_ALPHANUM12:
+        strToAssetCode(tl.asset.alphaNum12().assetCode, "USD12");
+        break;
+    default:
+        break;
+    }
+
+    if (Config::CURRENT_LEDGER_PROTOCOL_VERSION < 18 &&
```

Comment:
Added a comment

Reply:
I think I'm still not understanding this. What happens if I remove this check then build without `--enable-next-protocol-version-unsafe-for-production`?

Reply:
Then we would pass pool share trustlines around in v17 tests, causing them to fail. This is similar to the claimable balance `makeValid` call -  https://github.com/stellar/stellar-core/blob/c5f6349b240818f716617ca6e0f08d295a6fad9a/src/ledger/test/LedgerTestUtils.cpp#L250 I think the reason this doesn't fail at the moment is that we don't use generated entries in `for_version` blocks and then validate them.

It might make sense to introduce a layer between autocheck and the test cases where we pass the ledger version and make the entry valid for the current ledger version.

Reply:
Makes sense. Open an issue to remove the unnecessary checks (and maybe revisit this design)?

Reply:
https://github.com/stellar/stellar-core/issues/3185